pkg/build/registry/buildlog/rest.go

```diff
@@ -306,7 +306,10 @@ func (r *REST) Get(ctx apirequest.Context, name string, opts runtime.Object) (ru
 			}
 
 			containerLogOpts := buildapi.BuildToPodLogOptions(buildLogOpts)
-			containerLogOpts.Container = ""
+			//istio auto-inject may add sidecar containers to the build pod
+			//we are interested in getting logs from the build container only
+			containerLogOpts.Container = buildPod.Spec.Containers[0].Name
```

Comment:
i'm not totally confident we can rely on the ordering (someone could inject their container earlier in the list, theoretically).  It might be better to iterate looking for containers that match one of our expected names that we use when creating the build pod.

Reply:
ok, and are these the container names we are interested in?
```
pkg/build/controller/strategy/sti.go:83:        Name: "sti-build",
pkg/build/controller/strategy/custom.go:90:     Name: "custom-build",
pkg/build/controller/strategy/docker.go:60:     Name: "docker-build",
```

and would it be a good idea to group these in an enum type?

Reply:
yeah that should be the list.  having a constant list defined somewhere would make sense.

Reply:
(the downside to this approach of course is that we need to keep updating it if we add future images in the future)

Reply:
I've tried to group those container names together so hopefully in future, we see where to add the new container names